com.unity.multiplayer.mlapi/Tests/Runtime/NetworkObject/NetworkObjectDestroyTests.cs

```diff
@@ -0,0 +1,245 @@
+using System.Collections;
+using MLAPI.Configuration;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace MLAPI.RuntimeTests
+{
+    public class NetworkObjectDestroyTests
+    {
+        private const int k_ClientInstanceCount = 2;
+
+        private NetworkManager m_ServerNetworkManager;
+        private NetworkManager[] m_ClientNetworkManagers;
+        private GameObject m_PlayerPrefab;
+        private GameObject m_DummyPrefab;
+        private GameObject m_DummyGameObject;
+
+        [UnitySetUp]
+        public IEnumerator Setup()
+        {
+            // we need at least 1 client for tests
+            Assert.That(k_ClientInstanceCount, Is.GreaterThan(0));
+
+            Assert.That(MultiInstanceHelpers.Create(k_ClientInstanceCount, out m_ServerNetworkManager, out m_ClientNetworkManagers));
+            Assert.That(m_ServerNetworkManager, Is.Not.Null);
+            Assert.That(m_ClientNetworkManagers, Is.Not.Null);
+            Assert.That(m_ClientNetworkManagers.Length, Is.EqualTo(k_ClientInstanceCount));
+
+            m_PlayerPrefab = new GameObject("PlayerPrefabPrototype");
+            MultiInstanceHelpers.MakeNetworkedObjectTestPrefab(m_PlayerPrefab.AddComponent<NetworkObject>());
+
+            m_DummyPrefab = new GameObject("DummyPrefabPrototype");
+            MultiInstanceHelpers.MakeNetworkedObjectTestPrefab(m_DummyPrefab.AddComponent<NetworkObject>());
+
+            m_ServerNetworkManager.NetworkConfig.PlayerPrefab = m_PlayerPrefab;
+            m_ServerNetworkManager.NetworkConfig.NetworkPrefabs.Add(new NetworkPrefab { Prefab = m_DummyPrefab });
+            foreach (var clientNetworkManager in m_ClientNetworkManagers)
+            {
+                clientNetworkManager.NetworkConfig.PlayerPrefab = m_PlayerPrefab;
+                clientNetworkManager.NetworkConfig.NetworkPrefabs.Add(new NetworkPrefab { Prefab = m_DummyPrefab });
+            }
+
+            Assert.That(MultiInstanceHelpers.Start(/* isHost = */ true, m_ServerNetworkManager, m_ClientNetworkManagers));
```

Comment:
and, calling `Start(...)` with `false` (`isHost = false`) somehow makes `m_ClientNetworkManagers[0]` `null` and that's causing a `NullReferenceException` in tests when accessed (for instance, in a `foreach` loop).
there's also a bug in there potentially. @TwoTenPvP

Reply:
That does look like a bug indeed with the helper